scene/gui/scroll_container.h

```diff
@@ -50,7 +50,7 @@ class ScrollContainer : public Container {
 	HScrollBar *h_scroll = nullptr;
 	VScrollBar *v_scroll = nullptr;
 
-	Size2 child_max_size;
+	mutable Size2 child_max_size;
```

Comment:
What if this was calculated in `_update_dimensions` instead? I think that'd make more sense, and then we wouldn't need a `mutable` here (`mutable` = 😓)

Reply:
It needs to traverse all the child controls to calculate it.

The main purpose of `ScrollContainer::update_scrollbars()` is to confirm the status (`max`/`page`/`visible`) of scrollbars, it needs to have determined the largest one of the minimum size of child controls (`child_max_size`).

`ScrollContainer::_update_dimensions()` mainly determines the `rect` of the child controls. The `rect` needs to be calculated according to the `value` of the `ScrollBar`s.

So in terms of dependencies, it is unlikely to calculate all the data in one loop. Two loops are required. 

For convenience, it is better to calculate in `get_minimum_size()`. As the containers call `update_minimum_size()` before calling `queue_sort()`. See `*_child_notify(Node *p_child)` in `Container`.